tests/services/test_service_users.py

```diff
@@ -224,3 +238,55 @@ def test_deactivate(user_service, user_res, user_moderator):
         user_moderator.identity, q=f"username:{user_res._user.username}"
     )
     assert search.total > 0
+
+
+def test_non_existent_user_management(user_service, user_moderator):
+    """Try to manage a non-existent user."""
+    fake_user_id = 1000
+    funcs = [
+        user_service.block,
+        user_service.approve,
+        user_service.deactivate,
+        user_service.restore,
+    ]
+    for f in funcs:
+        with pytest.raises(PermissionDeniedError):
+            f(user_moderator.identity, fake_user_id)
+
+
+def test_restore(user_service, user_res, user_moderator):
+    """Test restore of a user."""
+    blocked = user_service.block(user_moderator.identity, user_res.id)
+    assert blocked
+
+    ur = user_service.read(user_moderator.identity, user_res.id)
+    assert ur.data["active"] == False
+    assert ur.data["blocked_at"] is not None
+
+    restored = user_service.restore(user_moderator.identity, user_res.id)
+    assert restored
+
+    ur = user_service.read(user_moderator.identity, user_res.id)
+    assert ur.data["active"] == True
+    assert ur.data["verified_at"] is not None
+    assert not "blocked_at" in ur.data
+
+
+def test_moderation_after_actions(user_service, user_res, user_moderator, monkeypatch):
+    """Test moderation actions (post block / restore)."""
+
+    def _block_action(user_id):
+        """Action to execute after blocking a user."""
+        # Uses system_identity since this will be running outside of the main request context
+
+        # This action is not useful at all, is just a way of testing that it is executed when blocking a user
+        user_service.deactivate(system_identity, user_id)
+
+    monkeypatch.setitem(current_actions_registry, "block", [_block_action])
```

Comment:
Since we are testing side effects of blocking a user, we add one action (in this case a silly one :) )